[MCPS] Add widgets toolset and tools to MCP Server docs (#36001)
DataDog/documentation: content/en/bits_ai/mcp_server/_index.md

- Add an agent trigger to the deployment rollback workflow so I can run it from here.
- Configure the incident response workflow to be triggerable by an agent.

### `get_widget`
*Toolset: **widgets***\

> *Permissions Required: `Dashboards Read` or `Timeseries Query` or `Monitors Read` or `APM Read` or `RUM Apps Read` (OR logic — any one permission grants access)*\

Reviewer: The `(OR logic — any one permission grants access)` parenthetical isn't present in any other tool entry — "or" already communicates this. Suggest removing it:

```suggestion
*Permissions Required: `Dashboards Read` or `Timeseries Query` or `Monitors Read` or `APM Read` or `RUM Apps Read`*\
```

> Retrieves and visualizes Datadog metrics, traces, logs, and other data as interactive charts. Supports three modes: dashboard lookup (provide a dashboard ID and widget ID), direct definition (provide a widget definition object), or URL (provide a Datadog share link or dashboard link with a focused widget). Returns formatted data with a sandbox URL for viewing the widget in Datadog.

Reviewer: Two issues on this line:

1. The opening sentence reads like a data query tool (closer to Metrics Explorer). `get_widget` works with widgets — leading with that sets the right expectation.
2. "Sandbox URL" isn't a standard Datadog term; suggest "preview URL" or a more descriptive phrase.

```suggestion
Retrieves and visualizes widget data from Datadog dashboards and notebooks as interactive charts. Supports three modes: dashboard lookup (provide a dashboard ID and widget ID), direct definition (provide a widget definition object), or URL (provide a Datadog share link or dashboard link with a focused widget). Returns formatted data with a preview URL for viewing the widget in Datadog.
```

- Show the CPU usage timeseries for `service:api` over the last hour.
- Fetch the widget data for widget `2228368921512806` on dashboard `abc-123-def`.
- Visualize the data from this Datadog share link: `https://app.datadoghq.com/s/yB5yjZ/h7d-srf-vm4`.

### `swap_widget_type`
*Toolset: **widgets***\
*Permissions Required: `Dashboards Read` or `Dashboards Write` or `Notebooks Read` or `Notebooks Write`*\

> Converts a widget definition from one visualization type to another while preserving queries. Supports formula-request-based widget types: timeseries, query_value, toplist, query_table, treemap, sunburst, distribution, heatmap, geomap, and list_stream.

Reviewer: "Formula-request-based" is internal API terminology that won't be meaningful to most readers. Suggest dropping it:

```suggestion
Converts a widget definition from one visualization type to another while preserving queries. Supports the following widget types: timeseries, query_value, toplist, query_table, treemap, sunburst, distribution, heatmap, geomap, and list_stream.
```

- Convert this timeseries widget to a toplist.
- Change the query table widget to a treemap visualization.

### `validate_notebook_cell`
*Toolset: **widgets***\
*Permissions Required: `Timeseries Query`*\

> Validates one or more notebook cell widget definitions, including SQL correctness for analysis_sql cells. When validating an analysis_sql cell, include its upstream data-source widgets so the endpoint can check SQL expressions against their schemas.

Reviewer: `analysis_sql` is a specific code identifier and should be formatted as inline code (both occurrences).

```suggestion
Validates one or more notebook cell widget definitions, including SQL correctness for `analysis_sql` cells. When validating an `analysis_sql` cell, include its upstream data-source widgets so the endpoint can check SQL expressions against their schemas.
```
